### PR TITLE
hps_accel: implement filter store registers

### PR DESCRIPTION
**Pull request by danc86**
This is largely a copy-paste from the input store implementation, with constants and memory depth tweaked, as a starting point. We expect the two implementations will start to diverge after this (particularly since the filter store will need to be shrunk to fit inside the fewer EBRAMs on HPS).

The tests in test_filter_store.py are independently implemented, which is why they look quite different to test_input_store.py.

This includes two x'ed out test cases which cover some edge cases around reset behaviour, which I think *should* work, but currently don't. It's probably not too important because we can easily avoid the problematic sequence of operations by just making sure we don't ever do that in software.

### CODE REVIEW

**Comment by danc86** [comment]
Amended to turn off writing traces in the new test cases.

File: proj/hps_accel/src/tensorflow/lite/kernels/internal/reference/integer_ops/conv_accel.cc [hunk 1]
```diff
@@ -81,7 +87,12 @@ void ConvPerChannel4x4(
         TFLITE_DCHECK_LE(in_x_origin + filter_width, input_width);
         const int8_t *current_input_data = input_data +
             Offset(input_shape, batch, in_y_origin, in_x_origin, 0);
+
+        TFLITE_DCHECK_LE(
+            input_depth * filter_height * filter_width / 4,
+            MAX_INPUT_WORDS);
```
Comment (alanvgreen) [inline proj/hps_accel/src/tensorflow/lite/kernels/internal/reference/integer_ops/conv_accel.cc:93]: Very nice.

File: proj/hps_accel/gateware/test_filter_store.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+            yield
+            # Now reset and fill again from the start
+            yield from self.reset_and_fill()
+            yield from self.read_out()
+        self.run_sim(process)
+
+    # TODO(dcallagh): fix reset logic so that this case passes.
+    # It currently breaks because the first output values get stuck at their
+    # output sources after the first fill, and remain stuck there during the
+    # second fill and up to the subsequent read.
```
Comment (alanvgreen) [inline proj/hps_accel/gateware/test_filter_store.py:172]: I believe the root cause of the bug is in SinglePortMemory. I have a rewrite in progress to simplify that class in order to more easily fix the bug.

**Review by alanvgreen — APPROVED** [review]
Looks good.

Love the test cases :)

Will merge now so I can make the changes needed to match the upcoming stream API PR.
